src/relations/s3.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024 Canonical Ltd.
```

Comment:
are these 2 files s3 and s3_helpers coming from somewhere? another charm?

Reply:
These are defined for this charm, but there are occurences of the helpers in other charms that relate to the s3 integrator as well (e.g. the `create_bucket_if_not_exists` method)